xbmc/platform/linux/input/LibInputKeyboard.cpp

```diff
@@ -28,159 +31,266 @@
-  { XKB_KEY_XF86AudioRecord, XBMCK_RECORD },
-  // XBMCK_REWIND clashes with XBMCK_MEDIA_REWIND
-  { XKB_KEY_XF86Phone, XBMCK_PHONE },
-  { XKB_KEY_XF86AudioPlay, XBMCK_PLAY },
-  { XKB_KEY_XF86AudioRandomPlay, XBMCK_SHUFFLE }
-  // XBMCK_FASTFORWARD clashes with XBMCK_MEDIA_FASTFORWARD
+static const std::unordered_map<xkb_log_level, int> logLevelMap = {
+    {XKB_LOG_LEVEL_CRITICAL, LOGERROR},  {XKB_LOG_LEVEL_ERROR, LOGERROR},
+    {XKB_LOG_LEVEL_WARNING, LOGWARNING}, {XKB_LOG_LEVEL_INFO, LOGINFO},
+    {XKB_LOG_LEVEL_DEBUG, LOGDEBUG},
 };
+
+static const std::unordered_map<xkb_keycode_t, XBMCKey> XkbDeadKeyXBMCMapping = {
+    {XKB_KEY_dead_grave, XBMCK_GRAVE},
+    {XKB_KEY_dead_tilde, XBMCK_TILDE},
+    {XKB_KEY_dead_acute, XBMCK_ACUTE},
+    {XKB_KEY_dead_circumflex, XBMCK_CIRCUMFLEX},
+    {XKB_KEY_dead_perispomeni, XBMCK_PERISPOMENI},
+    {XKB_KEY_dead_macron, XBMCK_MACRON},
+    {XKB_KEY_dead_breve, XBMCK_BREVE},
+    {XKB_KEY_dead_abovedot, XBMCK_ABOVEDOT},
+    {XKB_KEY_dead_diaeresis, XBMCK_DIAERESIS},
+    {XKB_KEY_dead_abovering, XBMCK_ABOVERING},
+    {XKB_KEY_dead_doubleacute, XBMCK_DOUBLEACUTE},
+    {XKB_KEY_dead_caron, XBMCK_CARON},
+    {XKB_KEY_dead_cedilla, XBMCK_CEDILLA},
+    {XKB_KEY_dead_ogonek, XBMCK_OGONEK},
+    {XKB_KEY_dead_iota, XBMCK_IOTA},
+    {XKB_KEY_dead_voiced_sound, XBMCK_VOICESOUND},
+    {XKB_KEY_dead_semivoiced_sound, XBMCK_SEMIVOICESOUND},
+    {XKB_KEY_dead_belowdot, XBMCK_BELOWDOT},
+    {XKB_KEY_dead_hook, XBMCK_HOOK},
+    {XKB_KEY_dead_horn, XBMCK_HORN},
+    {XKB_KEY_dead_stroke, XBMCK_STROKE},
+    {XKB_KEY_dead_abovecomma, XBMCK_ABOVECOMMA},
+    {XKB_KEY_dead_psili, XBMCK_ABOVECOMMA},
+    {XKB_KEY_dead_abovereversedcomma, XBMCK_ABOVEREVERSEDCOMMA},
+    {XKB_KEY_dead_dasia, XBMCK_OGONEK},
+    {XKB_KEY_dead_doublegrave, XBMCK_DOUBLEGRAVE},
+    {XKB_KEY_dead_belowring, XBMCK_BELOWRING},
+    {XKB_KEY_dead_belowmacron, XBMCK_BELOWMACRON},
+    {XKB_KEY_dead_belowcircumflex, XBMCK_BELOWCIRCUMFLEX},
+    {XKB_KEY_dead_belowtilde, XBMCK_BELOWTILDE},
+    {XKB_KEY_dead_belowbreve, XBMCK_BELOWBREVE},
+    {XKB_KEY_dead_belowdiaeresis, XBMCK_BELOWDIAERESIS},
+    {XKB_KEY_dead_invertedbreve, XBMCK_INVERTEDBREVE},
+    {XKB_KEY_dead_belowcomma, XBMCK_BELOWCOMMA},
+    {XKB_KEY_dead_a, XBMCK_DEAD_A},
+    {XKB_KEY_dead_A, XBMCK_DEAD_A},
+    {XKB_KEY_dead_e, XBMCK_DEAD_E},
+    {XKB_KEY_dead_E, XBMCK_DEAD_E},
+    {XKB_KEY_dead_i, XBMCK_DEAD_I},
+    {XKB_KEY_dead_I, XBMCK_DEAD_I},
+    {XKB_KEY_dead_o, XBMCK_DEAD_O},
+    {XKB_KEY_dead_O, XBMCK_DEAD_O},
+    {XKB_KEY_dead_u, XBMCK_DEAD_U},
+    {XKB_KEY_dead_U, XBMCK_DEAD_U},
+    {XKB_KEY_dead_small_schwa, XBMCK_SCHWA},
+    {XKB_KEY_dead_capital_schwa, XBMCK_SCHWA},
+};
+
+std::optional<XBMCKey> TranslateDeadKey(uint32_t keySym)
+{
+  auto mapping = XkbDeadKeyXBMCMapping.find(keySym);
+  return mapping != XkbDeadKeyXBMCMapping.end() ? std::optional<XBMCKey>(mapping->second)
+                                                : std::nullopt;
```

Comment:
Would this ever happen? Are there more mapings than are listed here?

Reply:
As far as I can tell I mapped all the deadkeys that exist on the keysym header so the likelihood is low. Regardless, this was the best approach I could find to wrap the translation is some reusable function while accounting for the inexistence of a key. Do you have any better suggestion?